src/backend/joanie/tests/core/test_api_admin_orders.py

```diff
@@ -810,3 +812,160 @@ def test_api_admin_orders_delete(self):
         response = self.client.delete(f"/api/v1.0/admin/orders/{order.id}/")
 
         self.assertEqual(response.status_code, HTTPStatus.METHOD_NOT_ALLOWED)
+
+    def test_api_admin_orders_cancel_anonymous(self):
+        """Anonymous user cannot cancel an order."""
+        order = factories.OrderFactory(
+            state=random.choice(
+                [
+                    enums.ORDER_STATE_CANCELED,
+                    enums.ORDER_STATE_SUBMITTED,
+                    enums.ORDER_STATE_DRAFT,
+                    enums.ORDER_STATE_PENDING,
+                    enums.ORDER_STATE_VALIDATED,
+                ]
+            )
+        )
+
+        response = self.client.post(f"/api/v1.0/admin/orders/{order.id}/cancel/")
+
+        self.assertEqual(response.status_code, HTTPStatus.UNAUTHORIZED)
+        content = response.json()
+        self.assertEqual(
+            content["detail"], "Authentication credentials were not provided."
+        )
```

Comment:
Maybe we can reduce noise, and only check the status code, as no custom response is send in json ?
```suggestion
        self.assertEqual(response.status_code, HTTPStatus.UNAUTHORIZED)
```
If you agree, the same can be done in other tests.